tinygrad/codegen/linearizer.py

```diff
@@ -45,6 +45,7 @@ def uop_alu_idx(self, a:UOp, b, ops, ctx:Linearizer, op, dtype=dtypes.int32):
 
   # NOTE: the consts have to be cached for deduping of downstream uops to work
   def const(self, b:Union[int,float], dtype=dtypes.int32, insert_before=None) -> UOp:
+    b = int(b) if dtypes.is_int(dtype) else float(b)
```

Comment:
This should be an assert and fixed elsewhere, it shouldn't be hotfixed here. The type should be correct, and if it isn't we should fix it where it isn't correct.

Reply:
I already posted this as an issue - will attempt to fix this

Reply:
Removed the cast as the issue no longer causes an error but that still persists, and there seem to be multiple places where the type is wrong, I'll try looking into it later